test/commit-validator.spec.js

```diff
@@ -1,93 +1,287 @@
+    });
+
+    it('contains warning if no leading blank line', async () => {
+      const { valid, errors, warnings } = await validator.validateCommit(
+        `
+        feat(bar): JIRA-1234 use new method
+        BREAKING CHANGE:
+        Use new method instead of old one.
+      `.trim(),
+      );
+
+      const footerLeadingBlankWarning = warnings.find(
+        (warning) => warning.name === 'footer-leading-blank',
+      );
+
+      expect(valid).toBe(true);
+      expect(errors).toHaveLength(0);
+      expect(warnings).toHaveLength(1);
+      expect(footerLeadingBlankWarning).toBeDefined();
+    });
+  });
+});
+
+describe('commit returns false if', () => {
+  it('contains invalid task', async () => {
+    const { valid, errors, warnings } = await validator.validateCommit(
+      'build(foo): JIRA-1234 Add build script',
     );
+
+    const typeError = errors.find((error) => error.name === 'type-enum');
+
+    expect(valid).toBe(false);
+    expect(errors).toHaveLength(1);
+    expect(warnings).toHaveLength(0);
+    expect(typeError).toBeDefined();
   });
 
-  it('returns false if message does not follow valid format', async () => {
-    const validator = new CommitValidator(options);
+  it('contains invalid scope', async () => {
+    const { valid, errors, warnings } = await validator.validateCommit(
+      'refactor(hello): JIRA-1234 Extract method',
+    );
+
+    const scopeError = errors.find((error) => error.name === 'scope-enum');
+
+    expect(valid).toBe(false);
+    expect(errors).toHaveLength(1);
+    expect(warnings).toHaveLength(0);
+    expect(scopeError).toBeDefined();
+  });
 
-    expect((await validator.validateCommit('refactor[foo]: JIRA-1234 Extract method')).valid).toBe(
-      false,
+  it('does not contain ticket number', async () => {
+    const { valid, errors, warnings } = await validator.validateCommit(
+      'refactor(foo): Extract method',
     );
-    expect((await validator.validateCommit('refactor(foo): JIRA-1234: Extract method')).valid).toBe(
-      false,
+
+    const subjectStartJiraError = errors.find((error) => error.name === 'subject-start-jira');
+
+    expect(valid).toBe(false);
+    expect(errors).toHaveLength(1);
+    expect(warnings).toHaveLength(0);
+    expect(subjectStartJiraError).toBeDefined();
+  });
+
+  it('contains invalid scope format', async () => {
+    const { valid, errors, warnings } = await validator.validateCommit(
+      'refactor[foo]: JIRA-1234 Extract method',
     );
-    expect((await validator.validateCommit('JIRA-1234: refactor(foo) Extract method')).valid).toBe(
-      false,
+
+    const scopeEmptyError = errors.find((error) => error.name === 'scope-empty');
+    const subjectEmptyError = errors.find((error) => error.name === 'subject-empty');
+    const typeEmptyError = errors.find((error) => error.name === 'type-empty');
+
+    expect(valid).toBe(false);
+    expect(errors).toHaveLength(3);
+    expect(warnings).toHaveLength(0);
```

Comment:
For empty arrays, it might be better to just compare to `[]`:

```suggestion
    expect(warnings).toEqual([]);
```